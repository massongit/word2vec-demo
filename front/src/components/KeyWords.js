import React from "react"
import {FormattedMessage} from "react-intl"
import PropTypes from "prop-types"

/**
 * positiveとnegativeのPropTypes
 * @type {{positive: shim, negative: shim}}
 */
export const propTypesPN = {
    positive: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    negative: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired
}

/**
 * キーワード
 */
class KeyWords extends React.Component {
    static propTypes = propTypesPN

    render() {
        return (
            <p>
                <label htmlFor="keywords">
                    <FormattedMessage id="header.keywords"/>
                </label>
                <p id="keywords">
                    {
                        (() => {
                            let r = []

                            if (this.props.positive) {
                                r = r.concat(this.props.positive.map(w => (
                                    "+" + w
                                )))
                            }

                            if (this.props.negative) {
                                r = r.concat(this.props.negative.map(w => (
                                    "-" + w
                                )))
                            }

                            return r.join(" ")
                        })()
                    }
                </p>
            </p>
        )
    }
}

export default KeyWords
