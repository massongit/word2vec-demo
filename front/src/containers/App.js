import App from "../components/App"
import {connect} from "react-redux"

const mapStateToProps = state => ({
    similar: state.showSimilarWords.words.similar
})

/**
 * ルート要素のContainer
 */
export default connect(mapStateToProps)(App)
