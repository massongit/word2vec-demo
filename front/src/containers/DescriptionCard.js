import DescriptionCard from "../components/DescriptionCard"
import {injectIntl} from "react-intl"

/**
 * 概要パネルのContainer
 */
export default injectIntl(DescriptionCard)
