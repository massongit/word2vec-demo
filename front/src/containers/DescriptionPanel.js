import DescriptionPanel from "../components/DescriptionPanel"
import {injectIntl} from "react-intl"

/**
 * 概要パネルのContainer
 */
export default injectIntl(DescriptionPanel)
