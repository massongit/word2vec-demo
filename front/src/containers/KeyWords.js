import KeyWords from "../components/KeyWords"
import {connect} from "react-redux"

const mapStateToProps = state => ({
    positive: state.showSimilarWords.words.positive,
    negative: state.showSimilarWords.words.negative
})

/**
 * キーワードのContainer
 */
export default connect(mapStateToProps)(KeyWords)
