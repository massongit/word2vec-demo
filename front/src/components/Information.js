import React from "react"
import {Table} from "reactstrap"
import {FormattedNumber, intlShape} from "react-intl"
import {propTypesSimilar} from "./App"

/**
 * 計算結果表示部
 */
class Information extends React.Component {
    static propTypes = {
        ...propTypesSimilar,
        intl: intlShape.isRequired
    }

    /**
     * ヘッダー部を描画する
     * @returns {node} ヘッダー部
     */
    renderHeader() {
        return (
            <tr>
                {
                    [
                        "header.candidate",
                        "header.word",
                        "header.cosine"
                    ].map((id, i) => ( // 見出しを上下左右中央揃えで表示
                        <th
                            key={i}
                            className="text-center"
                            style={{
                                "verticalAlign": "middle"
                            }}
                        >
                            {
                                this.props.intl.formatMessage({
                                    id
                                })
                            }
                        </th>
                    ))
                }
            </tr>
        )
    }

    /**
     * コンテンツ部を描画する
     * @returns {any[]} コンテンツ部
     */
    renderContents() {
        return this.props.similar.map((w, j) => (
            <tr
                key={j}
            >
                {
                    [
                        this.props.intl.formatNumber(j + 1),
                        w.word
                    ].map((data, i) => (
                        <td
                            className="text-center"
                            key={i}
                        >
                            {data}
                        </td>
                    ))
                }
                <td className="text-right">
                    <FormattedNumber
                        value={w.cosine}
                    />
                </td>
            </tr>
        ))
    }

    render() {
        return (
            <Table striped bordered hover size="sm" className="mb-0">
                <thead>
                    {this.renderHeader()}
                </thead>
                <tbody>
                    {this.renderContents()}
                </tbody>
            </Table>
        )
    }
}

export default Information
