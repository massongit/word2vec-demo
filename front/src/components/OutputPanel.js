import React from "react"
import PanelBody from "../containers/PanelBody"
import Information from "../containers/Information"
import {Panel, Row} from "react-bootstrap"
import {FormattedMessage} from "react-intl"
import KeyWords from "../containers/KeyWords"

/**
 * 計算結果表示部のパネル
 */
const OutputPanel = () => (
    <Row>
        <Panel>
            <Panel.Heading>
                <FormattedMessage id="title.result"/>
            </Panel.Heading>
            <PanelBody>
                <KeyWords/>
            </PanelBody>
            <Information/>
        </Panel>
    </Row>
)

export default OutputPanel
