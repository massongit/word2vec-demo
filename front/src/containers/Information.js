import Information from "../components/Information"
import {connect} from "react-redux"
import {injectIntl} from "react-intl"

const mapStateToProps = state => ({
    similar: state.showSimilarWords.words.similar
})

/**
 * 計算結果表示部のContainer
 */
export default injectIntl(connect(mapStateToProps)(Information))
