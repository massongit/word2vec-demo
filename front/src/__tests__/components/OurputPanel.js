import React from "react"
import OutputPanel from "../../components/OutputPanel"
import Information from "../../containers/Information"
import rootReducer from "../../reducers/index"
import {shallow} from "enzyme"
import {createStore} from "redux"
import {doSnapshot, functions} from "../containers/KeyWords"

let store

/**
 * テストの前処理
 */
const beforeProcess = () => (
    shallow(
        <OutputPanel
            store={store}
        />
    ).dive()
)

describe("containers/OutputPanel", () => {
    beforeEach(() => {
        store = createStore(rootReducer)
    })

    it("初期状態からStateが遷移した際に、Componentが正しく配置されている", () => {
        doSnapshot([functions], store, beforeProcess)
    })

    it("初期状態からStateが遷移した際に、子要素にInformationが含まれない", () => {
        functions(store)
        expect(beforeProcess().children().contains(<Information/>)).toBeTruthy()
    })
})
