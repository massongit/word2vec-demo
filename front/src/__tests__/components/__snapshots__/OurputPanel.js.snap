// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/KeyWords 初期状態からStateが遷移した際に、Componentが正しく配置されている 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <KeyWords
    dispatch={[Function]}
    negative={
        Array [
            "男",
          ]
    }
    positive={
        Array [
            "王",
            "女",
          ]
    }
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "unsubscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="keywords"
>
          <FormattedMessage
                    id="header.keywords"
                    values={Object {}}
          />
</label>,
        <p
          id="keywords"
>
          +王 +女 -男
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FormattedMessage
            id="header.keywords"
            values={Object {}}
/>,
          "htmlFor": "keywords",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "header.keywords",
            "values": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "+王 +女 -男",
          "id": "keywords",
        },
        "ref": null,
        "rendered": "+王 +女 -男",
        "type": "p",
      },
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="keywords"
>
            <FormattedMessage
                        id="header.keywords"
                        values={Object {}}
            />
</label>,
          <p
            id="keywords"
>
            +王 +女 -男
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FormattedMessage
              id="header.keywords"
              values={Object {}}
/>,
            "htmlFor": "keywords",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "header.keywords",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+王 +女 -男",
            "id": "keywords",
          },
          "ref": null,
          "rendered": "+王 +女 -男",
          "type": "p",
        },
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`containers/OutputPanel 初期状態からStateが遷移した際に、Componentが正しく配置されている 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Row
    bsClass="row"
    componentClass="div"
>
    <Uncontrolled(Panel)>
        <PanelHeading
            bsClass="panel"
            componentClass="div"
        >
            <FormattedMessage
                id="title.result"
                values={Object {}}
            />
        </PanelHeading>
        <InjectIntl(PanelBody)>
            <Connect(KeyWords) />
        </InjectIntl(PanelBody)>
        <InjectIntl(Connect(Information)) />
    </Uncontrolled(Panel)>
</Row>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Uncontrolled(Panel)>
        <PanelHeading
                bsClass="panel"
                componentClass="div"
        >
                <FormattedMessage
                        id="title.result"
                        values={Object {}}
                />
        </PanelHeading>
        <InjectIntl(PanelBody)>
                <Connect(KeyWords) />
        </InjectIntl(PanelBody)>
        <InjectIntl(Connect(Information)) />
</Uncontrolled(Panel)>,
      "className": "row",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <PanelHeading
            bsClass="panel"
            componentClass="div"
>
            <FormattedMessage
                        id="title.result"
                        values={Object {}}
            />
</PanelHeading>,
          <InjectIntl(PanelBody)>
            <Connect(KeyWords) />
</InjectIntl(PanelBody)>,
          <InjectIntl(Connect(Information)) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "panel",
            "children": <FormattedMessage
              id="title.result"
              values={Object {}}
/>,
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "title.result",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(KeyWords) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Uncontrolled(Panel)>
          <PanelHeading
                    bsClass="panel"
                    componentClass="div"
          >
                    <FormattedMessage
                              id="title.result"
                              values={Object {}}
                    />
          </PanelHeading>
          <InjectIntl(PanelBody)>
                    <Connect(KeyWords) />
          </InjectIntl(PanelBody)>
          <InjectIntl(Connect(Information)) />
</Uncontrolled(Panel)>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <PanelHeading
              bsClass="panel"
              componentClass="div"
>
              <FormattedMessage
                            id="title.result"
                            values={Object {}}
              />
</PanelHeading>,
            <InjectIntl(PanelBody)>
              <Connect(KeyWords) />
</InjectIntl(PanelBody)>,
            <InjectIntl(Connect(Information)) />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "panel",
              "children": <FormattedMessage
                id="title.result"
                values={Object {}}
/>,
              "componentClass": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "title.result",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(KeyWords) />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
